src/codegen/generate.ts

```diff
@@ -539,11 +580,52 @@ export default class ApiGenerator {
       // anyOf -> union
       return this.getUnionType(schema.anyOf, undefined, onlyMode);
     }
+    if (schema.discriminator) {
+      // parent schema -> union
+      const mapping = schema.discriminator.mapping || {};
+      return this.getUnionType(
+        Object.values(mapping).map((ref) => ({ $ref: ref })),
+        undefined,
+        onlyMode,
+      );
+    }
     if (schema.allOf) {
       // allOf -> intersection
```

Comment:
I think this comment should move down along with the old implementation of this, right?

Reply:
Where exactly should the comment be? An intersection type is still generated in the new implementation, so I think it's ok to leave the comment in its current position.

Reply:
ok, let's leave it here then